### ISSUE TITLE
Java 8 pom.xml doesn't depend on JSP API

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/google-cloud-eclipse#2456]
Non-maven projects have this in the runtime. 

```
  <dependencies>
    <!-- Compile/runtime dependencies -->
    <dependency>
      <groupId>javax.servlet</groupId>
      <artifactId>javax.servlet-api</artifactId>
      <version>3.1.0</version>
      <scope>provided</scope>
    </dependency>
    <dependency>
      <groupId>jstl</groupId>
      <artifactId>jstl</artifactId>
      <version>1.2</version>
    </dependency>
```

**Comment by chanseokoh** [comment]
Flex WAR projects don't have it either.

**Comment by chanseokoh** [comment]
Java 7 standard Maven doesn't have it either.

**Comment by patflynn** [comment]
What's the issue exactly?

**Comment by chanseokoh** [comment]
Some projects created by our wizards have the JSP JAR (jsp-api-2.1.jar for Java 7 and javax.servlet.jsp-api-2.3.0.jar for Java 8), while some projects don't.

**Comment by elharo** [comment]
And it's needed for a project that uses JSPs, which many do. The server runtime supplies this jar on the classpath automatically, but you can't compile JSPs or run them locally without it.

**Comment by chanseokoh** [comment]
@briandealwis would you fix #2455 at the same time?
